### PR TITLE
Use ValidatePubKey from sigstore/sigstore

### PR DESCRIPTION
**Pull request by haydentherapper**
This improves RSA, ECDSA, and ED25519 (no-op) validation for imported keys.

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

#### Ticket Link

Fixes

#### Release Note

```release-note
Improved validation for imported RSA keys
```

### CODE REVIEW

**Comment by haydentherapper** [comment]
Hm, looks like we might have an issue due to Windows - https://github.com/golang/go/issues/1108

**Comment by dlorenc** [comment]
> Hm, looks like we might have an issue due to Windows - [golang/go#1108](https://github.com/golang/go/issues/1108)

This is hitting a few things now... cc @cpanato any ideas what we should do here?

**Comment by cpanato** [comment]
> > Hm, looks like we might have an issue due to Windows - [golang/go#1108](https://github.com/golang/go/issues/1108)
> 
> This is hitting a few things now... cc @cpanato any ideas what we should do here?

maybe we can use that https://github.com/hashicorp/go-syslog but then we will need to patch upstream

**Comment by cpanato** [comment]
did a change in a fork https://github.com/cpanato/boulder/commit/efa97bd595c8bf6a2c59729103319138686e6c30 and updated our side and push this change to a branch on my cosign fork and run CI here: https://github.com/cpanato/cosign/pull/175

seems to compile and tests passed :) 

i can submit this patch to `boulder` and then if they approve/merge we can update, in meantime we can use the `replace` directive, i guess

**Comment by dlorenc** [comment]
Thanks! Let's do the fork for now:)

**Comment by haydentherapper** [comment]
Looks to be building successfully on Windows, just an intermittent KinD E2E error now.

**Comment by haydentherapper** [comment]
@cpanato Looks like there's another suggestion to remove the logging statement from a `core` package in `boulder`

**Comment by cpanato** [comment]
based on the feedback from the letsencrypt maintainer I did a follow-up https://github.com/letsencrypt/boulder/pull/6029 and did the testing on our side here https://github.com/sigstore/cosign/pull/1676 which compile and tests passed

**Review by znewman01 — APPROVED** [review]
LGTM modulo testing/build/CI issues, which you all seem to have under control now :)

File: go.mod
Comment (aarongable) [inline go.mod:318]: You could update this `replace` directive to point at https://github.com/cpanato/boulder@d354056fcaf943609ffe8e0f5e30c68ea110c520 instead of `@1d6951d8899d` to double-check that your PR https://github.com/letsencrypt/boulder/pull/6029 resolves everything

Comment (haydentherapper) [reply go.mod]: Updated, the Windows test is now passing!

**Comment by dlorenc** [comment]
Looks like we need codegen still, but Windows is passing!

**Comment by haydentherapper** [comment]
All tests are now passing, and the replace directive has been removed. PR is ready to go.

**Review by dlorenc — APPROVED** [review]

**Pull request merged**
